patches/minecraft/net/minecraft/world/level/block/SpreadingSnowyDirtBlock.java.patch

```diff
@@ -1,13 +1,25 @@
 --- a/net/minecraft/world/level/block/SpreadingSnowyDirtBlock.java
 +++ b/net/minecraft/world/level/block/SpreadingSnowyDirtBlock.java
-@@ -35,8 +_,10 @@
+@@ -10,7 +_,7 @@
+ import net.minecraft.world.level.block.state.BlockState;
+ import net.minecraft.world.level.lighting.LayerLightEngine;
+ 
+-public abstract class SpreadingSnowyDirtBlock extends SnowyDirtBlock {
++public abstract class SpreadingSnowyDirtBlock extends SnowyDirtBlock implements net.minecraftforge.common.extensions.IForgeSpreadingBlock{
+    protected SpreadingSnowyDirtBlock(BlockBehaviour.Properties p_56817_) {
+       super(p_56817_);
+    }
+@@ -35,9 +_,12 @@
  
     public void m_7455_(BlockState p_56819_, ServerLevel p_56820_, BlockPos p_56821_, Random p_56822_) {
        if (!m_56823_(p_56819_, p_56820_, p_56821_)) {
+-         p_56820_.m_46597_(p_56821_, Blocks.f_50493_.m_49966_());
 +         if (!p_56820_.isAreaLoaded(p_56821_, 1)) return; // Forge: prevent loading unloaded chunks when checking neighbor's light and spreading
-          p_56820_.m_46597_(p_56821_, Blocks.f_50493_.m_49966_());
++         if(net.minecraftforge.common.block.spreader.SpreadBehaviors.canSpread(p_56819_, net.minecraftforge.common.block.spreader.SpreaderType.REVERT))
++            p_56820_.m_46597_(p_56821_, net.minecraftforge.common.block.spreader.SpreadBehaviors.getSpreadState(p_56819_, p_56820_, p_56821_, net.minecraftforge.common.block.spreader.SpreaderType.REVERT));
        } else {
-+         if (!p_56820_.isAreaLoaded(p_56821_, 3)) return; // Forge: prevent loading unloaded chunks when checking neighbor's light and spreading
           if (p_56820_.m_46803_(p_56821_.m_7494_()) >= 9) {
++            this.spread(p_56819_, p_56820_, p_56821_, p_56822_, 4, 1); if(true) return;
```

Comment:
For the sake of better readability, please move the `if (true) return;` to the next line, and add an explanatory `// Forge: ` comment just in case. Also, space before opening parenthesis for an `if`-expression.